intlogger.go

```diff
@@ -353,32 +375,69 @@ func (l *intLogger) logPlain(t time.Time, name string, level Level, msg string,
 				}
 			}
 
-			var key string
-
+			// Convert the field key to a string.
 			switch st := args[i].(type) {
 			case string:
 				key = st
 			default:
 				key = fmt.Sprintf("%s", st)
 			}
 
-			if strings.Contains(val, "\n") {
-				l.writer.WriteString("\n  ")
-				l.writer.WriteString(key)
-				l.writer.WriteString("=\n")
-				writeIndent(l.writer, val, "  | ")
-				l.writer.WriteString("  ")
-			} else if !raw && needsQuoting(val) {
+			// Optionally apply the ANSI "dim" (faint) and "bold"
+			// SGR values to the key.
+			if l.fieldColor != ColorOff {
+				color := color.New(color.Faint, color.Bold)
+				key = color.Sprint(key)
+			}
+
+			// Values may contain multiple lines, and that format
+			// is preserved, with each line prefixed with a "  | "
+			// to show it's part of a collection of lines.
+			//
+			// Values may also neeq quoting, if not all the runes
+			// in the value string are "normal", like if they
+			// contain ANSI escape sequences.
+			switch {
+			case strings.Contains(val, "\n"):
+				key = "\n  " + key
+
+				valStrBuilder := &strings.Builder{}
+				valStrWriter := newWriter(valStrBuilder, ColorOff)
+
+				valStrBuilder.WriteString("\n")
+
+				if l.fieldColor != ColorOff {
+					color := color.New(color.Faint)
+					writeIndent(valStrWriter, val, color.Sprint("  | "))
+				} else {
+					writeIndent(valStrWriter, val, "  | ")
+				}
+				valStrWriter.Flush(level)
+
+				valStrBuilder.WriteString("  ")
+
+				val = valStrBuilder.String()
+			case !raw && needsQuoting(val):
+				val = strconv.Quote(val)
+			}
+
+			// If the key contains a newline, because the value itself
+			// contains newlines, then we pad with an extra space.
+			if !strings.Contains(key, "\n") {
```

Comment:
I'm confused why the key would contain newlines?

Reply:
To maintain compatibility with the test suite expectations when a field contains a multiline value, I need to prepend a `\n` to the key.

Removing this part:
https://github.com/hashicorp/go-hclog/blob/8c37f289569f64b785c37d55a85d284d39fcbb71/intlogger.go#L401-L402

I will get this error:

```console
$ go test -timeout 30s -run ^(TestLogger|TestLogger_leveledWriter|TestLogger_JSON)$ github.com/hashicorp/go-hclog -v
...
=== RUN   TestLogger/formats_multiline_values_nicely
    /Users/picat/Documents/GitHub/go-hclog/logger_test.go:176:
        	Error Trace:	logger_test.go:176
        	Error:      	Not equal:
        	            	expected: "[INFO]  test: this is test: who=programmer\n  why=\n  | testing\n  | and other\n  | pretty cool things\n  \n"
        	            	actual  : "[INFO]  test: this is test: who=programmer why=\n  | testing\n  | and other\n  | pretty cool things\n  \n"

        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,2 @@
        	            	-[INFO]  test: this is test: who=programmer
        	            	-  why=
        	            	+[INFO]  test: this is test: who=programmer why=
        	            	   | testing
        	Test:       	TestLogger/formats_multiline_values_nicely
...
```

Reply:
Ah, I see. You combined a previous call to write the newline and then the key into a string concat. To minimize allocations, best to leave it as 2 calls if you can.

Reply:
I'll try to change it to two calls.

Reply:
Thanks! Sorry for being so persnickety about this code, it has such an impact on so many things just have to be really careful with it.

Reply:
No problem at all, totally makes sense!

Reply:
Should be addressed with https://github.com/hashicorp/go-hclog/pull/118/commits/bcbe74d4b449b702e59de51a83aa67822e7396e2